### ISSUE TITLE
Challenge #21 -Exploring or machine/deep learning techniques to detect and track tropical cyclones

### ISSUE THREAD

**Issue by EsperanzaCuartero** [esowc/challenges_2020#5]
# Challenge #21 - Exploring machine/deep learning techniques to detect and track tropical cyclones 
> **Stream 2** - Machine-Learning and Artificial Intelligence

> **IMPORTANT**: this challenge is eligible to apply for cloud credits from the European Weather Cloud.

### Goal
From this project we expect to see a program (preferably based on Python3) that will use a machine/deep learning module to learn to recognize tropical cyclones of different intensities. 

The project will consist of two stages: 
1. learn to detect tropical cyclones and their intensities using historical information based on a set of satellite images and the BestTrack (https://www.ncdc.noaa.gov/ibtracs/) database/ 
2. test whether the algorithm is able to reliably detect tropical cyclones using real time information (forecasts) 

### Mentors and skills
* **Mentors:** @linusmagnusson @pmaciel
* **Skills required**
  * Machine / Deep learning

<hr>

### Challenge description
Tropical cyclones are among the most devastating weather systems on the planet, and accurate predictions of them are essential to make reliable warnings. For these warnings, the information about the track of the cyclone and its intensity is required and needs to be extracted from forecasts. 
From observations (mainly satellite images), tropical cyclones are manually detected, and the estimated positions and intensities are broadcasted and put into the IBTrACS (“BestTrack”) database. The classification of the cyclone intensity is in most of the cases based on the so called Dvorak technique. 

Numerical weather forecast models iteratively predict a range of variables in a 4-dimensional hypercube/tensor (3 spatial dimensions plus time). A challenge is to find tropical cyclone features in this vast amount of data and to extract the track of the feature. Currently, the applied tracking algorithms are based on the surface pressure, circulation further up in the atmosphere and the temperature in the core of the cyclone. The algorithms are configured from human experience.  
The current weather models have the capability to use the 3-dimensional output to simulate satellite images. This open up the possibility to use the Dvorak technique also on model output to classify the cyclones. While the Dvorak technique is based on human experience, we here seek to explore machine learning technique to recognize and classify tropical cyclones. 

We propose to explore face recognition techniques to 
  1. Train the algorithm to detect a tropical cyclone from a satellite image (real or simulated from analysis data). To train the data we will use the observed tropical cyclones from the BestTrack database to build up a library of past cyclones and match with satellite images. 

  2. In a second stage the outcome from the training period will be used on simulated satellite images from forecasts to track tropical cyclone features and classify the intensity. 

If successful, we can in the future consider to apply this technique to other meteorological features that leads to severe weather. 

More information
* [Use face-recognition algorithm to track tropical cyclones](https://docs.google.com/document/d/1PRYXkUbRj3BLtQ00mVuIR9nDkgUAaXFtLlYp3DKkEBc/edit?usp=sharing)

**Comment by itsmohitanand** [comment]
@jwagemann Is it possible to get the sample data, on which some algorithms could be tried?

**Comment by jwagemann** [comment]
Hi @melioristic , the new ESoWC round starts next week and then we will get back to you and respond to your question.
Thanks for your understanding!

**Comment by itsmohitanand** [comment]
> Hi @melioristic , the new ESoWC round starts next week and then we will get back to you and respond to your question.
> Thanks for your understanding!

Any update?

**Comment by EsperanzaCuartero** [comment]
Hi @melioristic, many thanks for your interest.
 ESoWC is to launch its third edition 2020 very soon. Please, follow all updates on twitter @esowc_ecmwf, #ESoWC2020 and visit the website esowc.ecmwf.int
Once the application period is open, we will respond in detail to all your questions.

**Comment by linusmagnusson** [comment]
Dear @melioristic, thanks for showing the interest in this project! We will create some sample data and hopefully make it available next week.

**Comment by itsmohitanand** [comment]
> Dear @melioristic, thanks for showing the interest in this project! We will create some sample data and hopefully make it available next week.

That sounds perfect.

**Comment by itsmohitanand** [comment]
I found this interesting paper and video which classifies galaxies, spiral and elliptical, wherein the concept of transfer learning is utilized making training faster and easier. 
https://arxiv.org/pdf/1812.02183.pdf
A wonderful video has also been created, 
https://www.youtube.com/watch?time_continue=92&v=8-jcf1TZNdA&feature=emb_logo

Suggestions from mentors are welcomed.

**Comment by linusmagnusson** [comment]
Thanks @melioristic for the interesting material. Sorry for the delay about the test data, I am working on the explanation of the included variables.

**Comment by xiaoya27** [comment]
Hi@linusmagnusson, this challenge looks really cool! as I was reading the expected outcome in your document, I have questions I would like to ask. The first programme looks like an identification and  classification problem, the idea is quite clear. On the second programme you mentioned " automatically detect tropical cyclones in forcast", does it mean that the model trained in the first prgramme by BestTrack database will be tested with model forcast's data, or does it mean I should build on top of prgramme1 a timeseries forcast model?  And when writing up the proposal, is it required to write for both programmes or just one would be enough?

**Comment by xiaoya27** [comment]
Thanks in adance for reading through my rather long questions:)

**Comment by ghost** [comment]
My initial thoughts for the task are: one could apply a relatively simple machine learning classification algorithm like the K-Nearest Neighbors and then apply a more complex deep learning one e.g. Convolutional Neural Nets :-)

**Comment by dangalea** [comment]
Hi @linusmagnusson @pmaciel @jwagemann,

I will be presenting a poster at EGU, given it goes ahead, on something [very similiar](https://meetingorganizer.copernicus.org/EGU2020/EGU2020-9870.html) that I've been working on as part of my PhD, with a paper on it currently in the works.

A few questions that I have regarding this challenge:
- Is the objective to check whether a TC is present in a particular timestep or to locate it and track it across timesteps? If the former, would something similiar to what I'll be presenting be acceptable? If the latter, could something like [this](https://papers.nips.cc/paper/6932-extremeweather-a-large-scale-climate-dataset-for-semi-supervised-detection-localization-and-understanding-of-extreme-weather-events) be desirable, or would an R-CNN be, where data from across timesteps is used?
- Is the aim to train the model with real satellite images or ones simulated from a model? Would such data be easily available?
- Is there some thinking that this Deep Learning model will be implemented into the operational model? If so, that would tie in well with my PhD as I am currently implemeting my Deep Learning model into the MetUM.
- If the Deep Learning model is successfully trained and/or deployed, would the be scope in writing up the results in a paper?

Thanks in advance for going through my questions.

**Comment by linusmagnusson** [comment]
Thanks for showing the interest @xiaoya27 ! On the second programme you mentioned " automatically detect tropical cyclones in forcast", does it mean that the model trained in the first prgramme by BestTrack database will be tested with model forcast's data, or does it mean I should build on top of prgramme1 a timeseries forcast model?
The idea is to use the position of the tropical cyclones and satellite images from the same times to train the model, and in the 2nd stage apply the model to simulated images from forecasts. As the forecasts are available with a time dimension, one can build a time-series from the individual detections of cyclones. No need for a time-series forecast model.

**Comment by linusmagnusson** [comment]
@dgaleareading , thanks for your questions! Your project sounds very interesting. Our current idea is to use the simulated satellite images from the ECMWF high-resolution analysis together with the BestTrack positions. It can also be considered to compare the results from simulated and real satellite images. A method to link different time instances to make a trajectory would be great.

In the first stage we are thinking of using it for evaluation purposes and also see if in the future the same method can be used for other atmospheric features (we start with tropical cyclones as they are "easy").

Of course are scientific papers always a part of the scope.

Best Regards,

Linus

**Comment by dangalea** [comment]
@linusmagnusson Thanks for your reply. I have currently produced a model that detects whether a TC is present in a section of the globe using 10m wind speed, MSLP, and vorticity at 3 levels. I'm fairly sure that my model can be re-trained using the appropriate data to get a model with an acceptable accuracy.

Is the aim ot locate the TC, i.e. drawing a bounding box around it or identify its centre, or just check whether a TC is present in the given image?

**Comment by jwagemann** [comment]
Join us for our LIVE @ecmwf Summer of Weather Code Ask Me Anything session on 1 April 2020 at 2 pm (CET) (tomorrow).

Get infos first hand from the #ESoWC2020 organisers, mentors and former #ESoWC participants.
➡️[Sign up](https://esowc.ecmwf.int/ama)

**Comment by itsmohitanand** [comment]
> Thanks @melioristic for the interesting material. Sorry for the delay about the test data, I am working on the explanation of the included variables.

Is it possible to have some of the training data or we should just go on with a conceptual proposal?

**Comment by Muggle-K** [comment]
Interesting project！
I happened to be working on something related to this project，how can I participate?

**Comment by linusmagnusson** [comment]
@melioristic and others. Sorry for the delay for the sample data, but we had to think about the formats. The attached csv file has data for tropical cyclones tracked by the ECMWF cyclone tracker in the ECMWF analysis. The columns lat_p and lon_p gives the position of the centre of the cyclone defined by the minimum pressure. The columns wind and pres gives the intensity in terms of maximum wind and minimum pressure. lat_w and lon_w is the position for the maximum wind (less relevant here) and the windrad the distance between the centre of the cyclone and the maximum wind. Finally the the code for the cyclone is in column name. The sample data is for 1 September 00 and 12UTC and 2 September 00UTC when hurricane Dorian (05L) was over the Bahamas a some other cyclones were active as well.

I also attach three simulated satellite images from ECMWF analysis. The idea of the project is to use these images and the csv file to train an algorithm to detect tropical cyclone features. The top left corner has the position 90N, 0E and the bottom-right corner 90S, 360E. For the project we will use approx 3 years of training data and 1 year of validation data. The data can also be provided in grib or netCDF format.
![ssd_20190902_12 grb](https://user-images.githubusercontent.com/60182608/78455733-fa516900-7697-11ea-9dcb-a6ad92f2ee09.png)
![ssd_20190901_12 grb](https://user-images.githubusercontent.com/60182608/78455734-fb829600-7697-11ea-9598-f06c4c81f8da.png)
![ssd_20190901_00 grb](https://user-images.githubusercontent.com/60182608/78455735-fc1b2c80-7697-11ea-91a7-eac87594385c.png)
[tc_an_sample.csv.txt](https://github.com/esowc/challenges_2020/files/4431910/tc_an_sample.csv.txt)

**Comment by jwagemann** [comment]
Only 4 days left to apply to be part of ECMWF Summer of Weather Code 2020. 
Application deadline: **Wednesday, 22 April 2020 at 23:59 (BST)**. 
Submit your proposal [here](https://esowc.commpla.com/form/application-form).
